src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -175,6 +191,19 @@ void executeQuery(
             "_shard_count", Block{{DataTypeUInt32().createColumnConst(1, shards), std::make_shared<DataTypeUInt32>(), "_shard_count"}});
         auto external_tables = context->getExternalTables();
 
+        /// We determine output stream sort properties by building a local plan (local because otherwise table could be unknown).
+        /// If no local shard exist for this cluster, no sort properties will be provided, c'est la vie.
+        SortDescription sort_description;
+        DataStream::SortMode sort_mode;
+        for (const auto & shard_info : query_info.getCluster()->getShardsInfo())
+        {
+            if (shard_info.isLocal())
```

Comment:
does this somehow rely on `order by` expression on the mergetree table? what if shards have different `order by` expressions? seen in production :)

Reply:
it is important observation, thx. current plan is to introduce smth like enforcer step in query planning which will be communicated to remote nodes and added to their query plans to guarantee sorting properties.